### PR TITLE
[#176140405] added impl functions activate and getstatus payment

### PR DESCRIPTION
**Pull request by pasqualespica**
### Description

The goal of this PR is to define the functions implementation related to :

- _ActivatePayment_.
See action **6** `/payment-activations {rptId, importoSingoloVersamento, CCP}` on _UML_ **Sequence diagram : Partale Pagamenti** present on main `README`

- _GetActivationStatus_
See action **10** `/payment-activations/{CCP}` on _UML_  **Sequence diagram : Partale Pagamenti** present on main `README`

### Proposed changes

Add the following API endpoints :
- **POST** `/payment-activations {rptId, importoSingoloVersamento, CCP}` where : 
    - `rtdId` is defined [here](https://docs.italia.it/italia/pagopa/pagopa-codici-docs/it/stabile/_docs/Capitolo2.html)
    - `CCP` is defined [here](https://docs.italia.it/italia/pagopa/pagopa-specifichepagamenti-docs/it/2.1/_docs/13-Capitolo_7/Capitolo7.html#codice-contesto-pagamento)

- **GET** `/payment-activations/{CCP}`

> **NOTE** : the `payment-activations` action causes the [debt position lock](https://anteprima-sanp.readthedocs.io/en/latest/_docs/SANP_2.2_Sez2_Cap02_GestionePosizioneDebitoria.html) even if payment is not made

### CODE REVIEW

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts [hunk 1]
```diff
@@ -1,11 +1,81 @@
...
+  };
+
+  const handler = ActivatePaymentHandler(apiClientMock as any);
+
+  const response = await handler(context, validPaymentActivationsRequest);
+
+  expect(response.kind).toBe("IResponseSuccessJson");
+});
+
+it("should return a PaymentActivationsPostResponse if the activation is successful", async () => {
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:65]: This is a negative case, isn't it? I think the descrition should be _... if the activation is **not** successful_

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts [hunk 2]
```diff
@@ -1,11 +1,81 @@
...
-    const httpHandler = HttpHandler();
-    const response = await httpHandler();
-    expect(response.kind).toBe("IResponseSuccessJson");
-  });
+import { Context } from "@azure/functions";
+import { right } from "fp-ts/lib/Either";
+import { PaymentActivationsPostRequest } from "../../generated/pagopa-proxy/PaymentActivationsPostRequest";
+import { ActivatePaymentHandler } from "../handler";
+
+export const context = ({
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:8]: Why the `context` is exported?

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts (hunk 1 above)
Comment (pasqualespica) [reply io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:65]: right

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts [hunk 3]
```diff
@@ -1,11 +1,81 @@
...
+    warn: jest.fn().mockImplementation(console.log)
+  }
+} as any) as Context;
+
+afterEach(() => {
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+});
+
+const validPaymentActivationsRequest = {
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:27]: Not a critical point, just an idea: wouldn't it be nice to collect all the _fake_ data in a dedicated module so as to allow random generation of test inputs or _fuzz testing_?

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts (hunk 2 above)
Comment (pasqualespica) [reply io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:8]: previously its part of external file under `__mock__` folder. i'll remove it. tnx

File: io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts [hunk 4]
```diff
@@ -1,11 +1,92 @@
...
-
-describe("HttpCtrl", () => {
-  it("should return a string when the query parameter is provided", async () => {
-    const httpHandler = HttpHandler();
-    const response = await httpHandler();
-    expect(response.kind).toBe("IResponseSuccessJson");
-  });
+import { GetActivationStatusHandler } from "../handler";
+
+// should be HERE !!!!
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts:5]: This comment is not so clear to me. I can't see _why_  `process.env` manipulation should be there. Maybe we should evaluate the use of a configuration file.

File: io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts [hunk 5]
```diff
@@ -1,11 +1,92 @@
...
+});
+
+it("should return a payment info KO response", async () => {
+  const apiClientMock = {
+    getActivationStatus: jest.fn(_ => {
+      return Promise.resolve(
+        left({
+          status: 400,
+          value: {
+            type: "bahhh"
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts:53]: :-) very strange type...

File: io-functions-pay-portal/ActivatePayment/handler.ts [hunk 6]
```diff
@@ -1,34 +1,71 @@
...
+        IResponseSuccessJson<PaymentActivationsPostResponse> | ErrorResponses
+      >(identity, identity)
+      .run();
   };
 }
 
-export function HttpCtrl(): express.RequestHandler {
-  const handler = HttpHandler();
-
+export function ActivatePaymentCtrl(
```
Comment (GiovanniMancini) [inline io-functions-pay-portal/ActivatePayment/handler.ts:65]: Maybe I'm wrong but theese factories of `RequestHandler` objects should be covered by tests, unless we consider them just boilerplate. I mean.. we could test the resulting express application

**Review by GiovanniMancini — APPROVED** [review]

File: io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts (hunk 3 above)
Comment (pasqualespica) [reply io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts:27]: yes

File: io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts (hunk 4 above)
Comment (pasqualespica) [reply io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts:5]: you are right, I haven't investigated so much, the problem is that is used into `getConfigOrThrow` by `pagopa.ts` client. one more thing to review, tnx 👍

File: io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts (hunk 5 above)
Comment (pasqualespica) [reply io-functions-pay-portal/GetActivationStatus/__tests__/handler.test.ts:53]: yes with @infantesimone  we thought a lot about this type 🤪 . i'll change it

**Pull request merged**